### PR TITLE
Implement signing functionality in the orchestrator

### PR DESCRIPTION
**Pull request by jul-sh**
Implements basic signing functionality in the orchestrator, using the instance key.

### CODE REVIEW

**Review by ipetr0v — APPROVED** [review]

File: oak_containers_orchestrator/src/crypto.rs [hunk 1]
```diff
@@ -126,6 +125,22 @@ impl CryptoService {
     pub(crate) fn new(key_store: Arc<KeyStore>) -> Self {
         Self { key_store }
     }
+
+    fn siging_key(&self, key_origin: KeyOrigin) -> Result<&p256::ecdsa::SigningKey, tonic::Status> {
```
Comment (tiziano88) [inline oak_containers_orchestrator/src/crypto.rs:129]:
```suggestion
    fn signing_key(&self, key_origin: KeyOrigin) -> Result<&p256::ecdsa::SigningKey, tonic::Status> {
```

File: oak_containers_orchestrator/src/crypto.rs [hunk 2]
```diff
@@ -158,9 +173,16 @@ impl OrchestratorCrypto for CryptoService {
...
     async fn sign(
         &self,
-        _request: Request<SignRequest>,
+        request: Request<SignRequest>,
     ) -> Result<Response<SignResponse>, tonic::Status> {
-        // TODO(#4074): Implement once the DICE signing key is available in the orchestrator.
-        Err(tonic::Status::unimplemented("not yet implemented"))
+        let request = request.into_inner();
+        let signing_key = self.siging_key(request.key_origin())?;
+        let signature = <p256::ecdsa::SigningKey as oak_crypto::signer::Signer>::sign(
```
Comment (tiziano88) [inline oak_containers_orchestrator/src/crypto.rs:180]: Why is this type conversion needed?

**Review by tiziano88 — APPROVED** [review]
Great!

File: oak_containers_orchestrator/src/crypto.rs (hunk 2 above)
Comment (jul-sh) [reply oak_containers_orchestrator/src/crypto.rs:180]: It's not technically a type conversion, we're just using the specific `oak_crypto::signer::Signer` trait. 

`p256::ecdsa::SigningKey` implements multiple traits that provide a `sign` method. We could also put the trait into the imports as usual, but given it implements multiple sign traits I think this clearer.

File: oak_containers_orchestrator/src/crypto.rs (hunk 1 above)
Comment (jul-sh) [reply oak_containers_orchestrator/src/crypto.rs:129]: haha thanks

**Review by conradgrobler — APPROVED** [review]
Thanks!

**Review by nfallen — APPROVED** [review]

File: oak_containers_orchestrator/src/crypto.rs (hunk 2 above)
Comment (tiziano88) [reply oak_containers_orchestrator/src/crypto.rs:183]: Thanks for clarifying

**Review by tiziano88 — APPROVED** [review]

**Pull request merged**